File: backend-python/validations/matches.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, Field


class ListMatcheQuery(BaseModel):
    limit: Optional[int] = Field(default=50, ge=1)

class CreateMatchPayload(BaseModel):
    sport: str = Field()
    home_team: str
    away_team: str
    start_time: datetime
    end_time: datetime
    home_score: Optional[int]
    away_score: Optional[int]

class UpdateScorePayload(BaseModel):
    home_score: int
    away_score: int

File: backend/src/validation/matches.rs
use chrono::{DateTime, Utc};
use serde::Deserialize;
use validator::Validate;

#[derive(Deserialize)]
pub struct ListMatchQuery {
    pub limit: Option<u64>,
}

#[derive(Deserialize)]
pub struct CreateMatchPayload {
    pub sport: String,
    pub home_team: String,
    pub away_team: String,
    pub start_time: DateTime<Utc>,
    pub end_time: DateTime<Utc>,
    pub home_score: Option<i32>,
    pub away_score: Option<i32>,
}

#[derive(Deserialize, Validate)]
pub struct UpdateScorePayload {
    #[validate(range(min = 0))]
    pub home_score: i32,
    #[validate(range(min = 0))]
    pub away_score: i32,
}